pytest_doctestplus/plugin.py

```diff
@@ -51,6 +51,23 @@ def __exit__(self, *args, **kwargs):
 """.lstrip()
 
 
+@pytest.fixture(autouse=True)
+def _docdir(request):
+    """Run doctests in isolated tmpdir so outputs do not end up in repo"""
+    # Trigger ONLY for doctestplus
+    doctest_plugin = request.config.pluginmanager.getplugin("doctestplus")
+    if isinstance(request.node, doctest_plugin._doctest_textfile_item_cls):
+        # Don't apply this fixture to io.rst.  It reads files and doesn't write
+        if "io.rst" not in request.node.name:
```

Comment:
`io.rst` sounds awfully specific to `astropy` core library. How do we generalize this?

Reply:
Only based on the narrative, I don't get why would it hurt to still apply when it reads files but not writes?

Reply:
Yeah, this is specific to the astropy core library.  The problem with that doctest is that it looks for an input file in the working directory.  If one makes the working dir a `tmpdir`, then the file is no longer in the working directory, so one needs a general way of getting the file.  Using `importlib.resources` might be the solution.

Reply:
Hmm looks like [importlib.resources is only available since Python 3.7](https://docs.python.org/3/library/importlib.html#module-importlib.resources).

Reply:
It is available as an external package for python 3.6 and earlier.  `importlib_resources`  One can do something like.

https://github.com/spacetelescope/asdf/blob/522f25b60715612dd1369a644c756fb7572b89d1/asdf/tests/helpers.py#L45-L51